src/Workspaces/CSharp/Portable/Extensions/SemanticModelExtensions.cs

```diff
@@ -445,5 +477,103 @@ private static TypeSyntax GetOutermostType(TypeSyntax type)
         {
             return type.GetAncestorsOrThis<TypeSyntax>().Last();
         }
+
+
+        // API that  gets semantic info for a specific arguments from @dustinca
+        //https://github.com/DustinCampbell/CSharpEssentials/blob/master/Source/CSharpEssentials/Extensions.cs
+        // supposedly there is a bug on github somewhere for him to add this api to the compiler
+        internal struct ArgumentInfo
+        {
+            public readonly ISymbol MethodOrProperty;
+            public readonly IParameterSymbol Parameter;
+
+            public ArgumentInfo(ISymbol methodOrProperty, IParameterSymbol parameter)
+            {
+                this.MethodOrProperty = methodOrProperty;
+                this.Parameter = parameter;
+            }
+        }
+
+        public static ArgumentInfo GetArgumentInfo(this SemanticModel semanticModel, ArgumentSyntax argument)
```

Comment:
I'm pretty certain there is an IOperation call that gives you this information directly.  Ask @JohnHamby if that's true.